packages/snaps-utils/src/handlers.ts

```diff
@@ -142,6 +144,29 @@ export const OnHomePageResponseStruct = union([
   OnHomePageResponseWithContentStruct,
   OnHomePageResponseWithIdStruct,
 ]);
+
+export const AddressResolutionStruct = object({
+  protocol: string(),
+  resolvedDomain: string(),
+});
+
+export const DomainResolutionStruct = object({
+  protocol: string(),
+  resolvedAddress: string(),
+});
+
+export const AddressResolutionResponseStruct = object({
+  resolvedDomains: size(array(AddressResolutionStruct), 1, Infinity),
+});
+
+export const DomainResolutionResponseStruct = object({
+  resolvedAddresses: size(array(DomainResolutionStruct), 1, Infinity),
```

Comment:
Do we need to enforce a non-empty array here? Doesn't seem to be specified in the SIP.

Reply:
I can update the SIP to be more specific, but yes this should be non-empty, otherwise they could have just returned `null`

Reply:
Let's mention it in the SIP then!

Reply:
Ok will update